src/Microsoft.TestPlatform.Common/PublicAPI/PublicAPI.Shipped.txt

```diff
@@ -20,9 +21,10 @@ Microsoft.VisualStudio.TestPlatform.Common.Exceptions.InvalidLoggerException
 Microsoft.VisualStudio.TestPlatform.Common.Exceptions.InvalidLoggerException.InvalidLoggerException(string message) -> void
 Microsoft.VisualStudio.TestPlatform.Common.Exceptions.InvalidLoggerException.InvalidLoggerException(string message, System.Exception innerException) -> void
 Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.DataCollectorMetadata
-Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.DataCollectorMetadata.DataCollectorMetadata(string extension, string friendlyName) -> void
+Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.DataCollectorMetadata.DataCollectorMetadata(string extension, string friendlyName, bool hasAttachmentProcessor) -> void
```

Comment:
This would break the exitsting public api. Is there a reason why this is okay?

Reply:
DataCollectorMetadata is an internal type for the datacollector extensione...I don't expect external usage outside platform. I don't see usage around https://grep.app/search?q=DataCollectorMetadata&regexp=true http://index/?query=DataCollectorMetadata
The way the metadata object is created is by convention(Type, metadata) 

https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs#L161

I don't see back-compat issue here, but if you're concerned about it we can keep old ctor and set false there. That value is used to have better performance during attachment processor creation, if hasAttachmentProcessor is false we skip dll loading/inspection.

Reply:
Overload added